src/vars/foreman.yml

```diff
@@ -2,3 +2,7 @@
 foreman_admin_passwd_file: "{{ obsah_state_path }}/foreman-admin-init-passwd"
 foreman_initial_admin_username: admin
 foreman_initial_admin_password: "{{ lookup('ansible.builtin.password', foreman_admin_passwd_file, chars=['ascii_letters']) }}"
+# Puma workers calculation:
+# CPU based calculation is based on https://github.com/puma/puma/blob/master/docs/deployment.md#mri
+# Memory based calculation is based on https://docs.gitlab.com/ee/install/requirements.html#puma-settings
+foreman_service_puma_workers: "{{ [100, ansible_processor_nproc * 1.5, (ansible_memtotal_mb / 1024) - 1.5] | min | int }}"
```

Comment:
The 100 is taken from Puppet which was added by @ekohl in https://github.com/theforeman/puppet-foreman/pull/1197, but https://github.com/puma/puma/blob/main/docs/deployment.md says nobody should do more than 32…

Reply:
That 100 was just a number that didn't break the system. Guess I should have read the docs back then. Sometimes it's good to revisit all the values

Reply:
I think I'd go with 32 then, allowing the user to go higher on their own if they really want to.

Reply:
In Ruby you have `Etc.nproc` to get the number of effective processors. Should we put that in Foreman's `config/environments/production.rb` as a default, leaving it empty here? Then it should auto tune, also taking quotas into account.

Reply:
> In Ruby you have `Etc.nproc` to get the number of effective processors. Should we put that in Foreman's `config/environments/production.rb` as a default, leaving it empty here? Then it should auto tune, also taking quotas into account.

You say putting the whole calculation into `config/puma/production.rb`? As just using `Etc.nproc` would be wrong.
works for me!

> 100 is static here is it suppose to be dynamic for different tuning profiles. and yes we should keep it to 32 as per docs

The 100 is purely a "magic number" for "not too many". I think even in the tuning profiles we wouldn't go over 32 by default.

Reply:
A quick look shows me there are implementations but they (on Linux) parse `/proc/meminfo` or shell out. Not pretty. Either it's not portable or a lot of code. Keeping it here is fair.

Reply:
Yepp, that's also what I had found.